sdks/java/io/kinesis/src/main/java/org/apache/beam/sdk/io/kinesis/ShardRecordsIterator.java

```diff
@@ -43,22 +42,23 @@
   private final RecordFilter filter;
   private final String streamName;
   private final String shardId;
-  private AtomicReference<ShardCheckpoint> checkpoint;
+  private final AtomicReference<ShardCheckpoint> checkpoint;
+  private final WatermarkPolicy watermarkPolicy;
```

Comment:
Why `AtomicReference` no more used here?

Reply:
I don't know why it was used in the first place, maybe it was some leftover after some previous changes to the code. The value of `watermarkPolicy` is only set in constructor and then only accessed (never changed), so there's no need to have an `AtomicReference` for that. From the perspective of memory visiblity, making the field as `final` ensures that the value of the field is correctly published to other threads, so there's no difference also in that aspect.